### ISSUE TITLE
delete header in userResHeaderDecorator

### ISSUE THREAD

**Issue by cuongluu8** [villadora/express-http-proxy#453]
Is it possible to delete a header in `userResHeaderDecorator`?
I can set it to null or "" by doing `headers["x-my-secret-header"] = "";`, but ideally I wouldn't want this header to be surfaced in the browser at all.

`delete headers["x-my-secret-header"];` doesn't work.

UPDATE:
```
const {
  "x-my-secret-header": ignore, // eslint-disable-line no-unused-vars
  ...headersToReturn
} = headers;

return headersToReturn;
```
so `headersToReturn` will not contain this header, but it seems that it will still be returned in the actual response.

**Comment by cuongluu8** [comment]
So it seems that you also need to explicitly remove the header from the `userRes`
`userRes.removeHeader("x-my-secret-header");`

Not sure if this is expected behaviour or a bug.

**Comment by monkpow** [comment]
Thanks for the report @cuongluu8. As you describe it, this is unexpected. Let me write a couple of tests and see if I can reproduce.

**Comment by monkpow** [comment]
This is a defect. I am able to reproduce and have a failing test. I should be able to fix this shortly. Thanks again for the report.

**Comment by monkpow** [comment]
I believe this is closed with #461
